### ISSUE TITLE
Question Regarding the Role of S_{EMA} in the Advantage Computation

### ISSUE THREAD

**Issue by Quarkstar** [NVlabs/RLP#3]
Hello,

I have a question about the relationship between the information-gain reward and the group-relative baseline described in your paper. I've been looking at the equations, and I would be very grateful for some clarification.

The paper defines the information-gain reward as:

$$
r(c_t) := S_{\text{pred}}(c_t) - S_{\text{EMA}}
$$

And the advantage using a group-relative baseline is defined as:

$$
A^{(i)} := \frac{G}{G-1}\Big(r(c_t^{(i)}) - \bar r\Big), \quad \text{where} \quad \bar r := \frac{1}{G}\sum_{j=1}^G r(c_t^{(j)})
$$

I wanted to understand the final form of the advantage, so I tried substituting the definition of the reward into the advantage equation.

First, expanding the average reward $\bar r$:

$$
\begin{align*}
\bar r &= \frac{1}{G}\sum_{j=1}^G r(c_t^{(j)}) \\
&= \frac{1}{G}\sum_{j=1}^G \Big(S_{\text{pred}}(c_t^{(j)}) - S_{\text{EMA}}\Big) \\
&= \left(\frac{1}{G}\sum_{j=1}^G S_{\text{pred}}(c_t^{(j)})\right) - S_{\text{EMA}}
\end{align*}
$$

The $S_{\text{EMA}}$ term is constant across all $j=1, \dots, G$ for a given next token $x_t$.

Next, substituting both $r(c_t^{(i)})$ and the expanded $\bar r$ back into the advantage formula $A^{(i)}$:

$$
\begin{align*}
A^{(i)} &:= \frac{G}{G-1}\Big(r(c_t^{(i)}) - \bar r\Big) \\
&= \frac{G}{G-1} \left[ \left(S_{\text{pred}}(c_t^{(i)}) - S_{\text{EMA}}\right) - \left(\left(\frac{1}{G}\sum_{j=1}^G S_{\text{pred}}(c_t^{(j)})\right) - S_{\text{EMA}}\right) \right] \\
&= \frac{G}{G-1} \left[ S_{\text{pred}}(c_t^{(i)}) - S_{\text{EMA}} - \frac{1}{G}\sum_{j=1}^G S_{\text{pred}}(c_t^{(j)}) + S_{\text{EMA}} \right] \\
\end{align*}
$$

The two $S_{\text{EMA}}$ terms cancel each other out, leading to the final expression:

$$
A^{(i)} = \frac{G}{G-1} \left( S_{\text{pred}}(c_t^{(i)}) - \frac{1}{G}\sum_{j=1}^G S_{\text{pred}}(c_t^{(j)}) \right)
$$

Based on this derivation, the final formula for the advantage $A^{(i)}$ does not appear to contain the $S_{\text{EMA}}$ term. This seems to imply that the "no-think baseline" is not actually used in the policy update when the group-relative baseline is applied.

Could you please clarify if my derivation is correct? If it is, I may be misunderstanding the conceptual role of $S_{\text{EMA}}$. Is there another aspect of the optimization or a subtlety I'm missing where $S_{\text{EMA}}$ has an effect?

Thank you for your time and for the excellent work.

**Comment by tvergara** [comment]
Hi, in our lab we crossed into this same issue. I believe that, as currently explained in the paper, the $S_{\text{EMA}}$ is not actually affecting the advantages at all. @ahatamiz can you help us understand this part?

**Comment by ahatamiz** [comment]
Hi @Quarkstar @tvergara 

Within a single group of G sampled thoughts for the same context, the mean advantage removes any term that is constant so the per-group update direction does not depend on S_EMA. But the S_EMA is crucial for the method's guarantees. Without it, it is very easy to pose the optimization into reward-hacking.

You can think about the S_EMA as an anchor that makes the training stable by decoupling the reward from the actor. Also, although the within-group update cancels any constants, we still use the raw reward r(c_t) and the group mean r_mean to reason about when thinking helps. If r_mean<0 at a position, thinking hurts relative to the EMA and the policy would learn to emit the stop token quickly because only shorter thoughts can become above average in that group.

Here's also an analogy that may better help understand the purpose of S_EMA. Let's say we have n students with different capabilities. In case 1, we ask them to take an exam and grade themselves. The door is open for cheating as they can give the highest grade to themselves but that does change anything about how each student compare to each other (some are really good and some may not). In case 2, we have a teacher that grades them. This would avoid the cheating issue. But this does not change the way students compare to each other as their skill-sets are the same (of course assuming that grading is done in fair manner that does not penalize anyone). S_EMA in this case is the teacher that works as an anchor. Without it, the policy collapses easily.

Lastly, we present two formal guarantees that rely on S_EMA. In proposition 1, we prove that the expected reward in info gain is exactly the reduction in cross-entropy. If we remove S_EMA, we no longer measure the improvement of thinking vs no-thinking, so we lose the guarantee. Also In proposition 2, we show that maximizing CoT-conditioned objective increases a computable lower bound on how much the predictor improves over the non-think baseline. Similarly, if we remove S_EMA, there is nothing to bound against.

I hope this addresses your question. 

Kind Regards,
